module_8_1
    Домашнее задание по уроку "Try и Except".
    
    Задание "Программистам всё можно":
    Пора разрушать шаблоны привычного нам Python! Вот вас раздражает, что мы не можем сложить число(int) и строку(str)? Давайте исправим это недоразумение!
    
    Реализуйте следующую функцию:
    add_everything_up, будет складывать числа(int, float) и строки(str)
    
    Описание функции:
    add_everything_up(a, b) принимает a и b, которые могут быть как числами(int, float), так и строками(str).
    TypeError - когда a и b окажутся разными типами (числом и строкой), то возвращать строковое представление этих двух данных вместе (в том же порядке). Во всех остальных случаях выполнять стандартные действия.
    
    Пример кода:
    print(add_everything_up(123.456, 'строка'))
    print(add_everything_up('яблоко', 4215))
    print(add_everything_up(123.456, 7))
    
    Вывод в консоль:
    123.456строка
    яблоко4215
    130.456
    
    Примечания:
    Конструкции try-except в функции выполняют строго ту обработку, которая написана в задании (обращайте на важность порядка передачи данных).
    Если хотите дополнить функции своими исключениями или написать отдельно дополнительно собственные функции - это не запрещено, мы не против, чтобы вы больше экспериментировали. Но в первую очередь выполните поставленную задачу.

module_8_2
    Цель: понять как работают исключения внутри функций и как обрабатываются эти исключения на практике при помощи try-except.
    
    Задача "План перехват":
    Напишите 2 функции:
    Функция personal_sum(numbers):
    Должна принимать коллекцию numbers.
    Подсчитывать сумму чисел в numbers путём перебора и увеличивать переменную result.
    Если же при переборе встречается данное типа отличного от числового, то обработать исключение TypeError, увеличив счётчик incorrect_data на 1.
    В конечном итоге функция возвращает кортеж из двух значений: result - сумма чисел, incorrect_data - кол-во некорректных данных.
    Функция calculate_average(numbers)
    Среднее арифметическое - сумма всех данных делённая на их количество.
    Должна принимать коллекцию numbers и возвращать: среднее арифметическое всех чисел.
    Внутри для подсчёта суммы используйте функцию personal_sum написанную ранее.
    Т.к. коллекция numbers может оказаться пустой, то обработайте исключение ZeroDivisionError при делении на 0 и верните 0.
    Также в numbers может быть записана не коллекция, а другие типы данных, например числа. Обработайте исключение TypeError выводя строку 'В numbers записан некорректный тип данных'. В таком случае функция просто вернёт None.
    
    Пункты задачи:
    Создайте функцию personal_sum на основе условий задачи.
    Создайте функцию calculate_average на основе условий задачи.
    Вызовите функцию calculate_average несколько раз, передав в неё данные разных вариаций.
    Пример результата выполнения программы:
    Пример выполнения программы:
    print(f'Результат 1: {calculate_average("1, 2, 3")}') # Строка перебирается, но каждый символ - строковый тип
    print(f'Результат 2: {calculate_average([1, "Строка", 3, "Ещё Строка"])}') # Учитываются только 1 и 3
    print(f'Результат 3: {calculate_average(567)}') # Передана не коллекция
    print(f'Результат 4: {calculate_average([42, 15, 36, 13])}') # Всё должно работать
    
    Вывод на консоль:
    Некорректный тип данных для подсчёта суммы - 1
    Некорректный тип данных для подсчёта суммы - ,
    Некорректный тип данных для подсчёта суммы -
    Некорректный тип данных для подсчёта суммы - 2
    Некорректный тип данных для подсчёта суммы - ,
    Некорректный тип данных для подсчёта суммы -
    Некорректный тип данных для подсчёта суммы - 3
    Результат 1: 0
    Некорректный тип данных для подсчёта суммы - Строка
    Некорректный тип данных для подсчёта суммы - Ещё Строка
    Результат 2: 2.0
    В numbers записан некорректный тип данных
    Результат 3: None
    Результат 4: 26.5
